### ISSUE TITLE
JQuery UI loading performance

### ISSUE THREAD

**Issue by onlinebizsoft** [magento/magento2#36923]
### Summary

Magento improved from time to time but seems to break performance again after a recent update

https://github.com/magento/magento2/issues/32810 - we removed some unnecessary dependencies from this ticket
https://github.com/magento/magento2/commit/fa87f9299985cf0b7d1b778f056bfceff1aec231 - after an upgrade, dialog.js get wrong again 

### Examples

https://github.com/magento/magento2/blob/fa87f9299985cf0b7d1b778f056bfceff1aec231/lib/web/jquery/ui-modules/widgets/dialog.js

### Proposed solution

We should review and remove dependencies from here again
![image](https://user-images.githubusercontent.com/22501570/221478323-a36fb873-8142-4775-abae-512ab3e75bb7.png)

### Triage and priority

- [ ] Severity: **S0** _- Affects critical data or functionality and leaves users without workaround._
- [ ] Severity: **S1** _- Affects critical data or functionality and forces users to employ a workaround._
- [ ] Severity: **S2** _- Affects non-critical data or functionality and forces users to employ a workaround._
- [X] Severity: **S3** _- Affects non-critical data or functionality and does not force users to employ a workaround._
- [ ] Severity: **S4** _- Affects aesthetics, professional look and feel, “quality” or “usability”._

**Comment by onlinebizsoft** [comment]
@mrtuvn can you check and confirm?

**Comment by engcom-Hotel** [comment]
Hello @onlinebizsoft,

Thanks for the report and collaboration!

As per the description of this issue as well as https://github.com/magento/magento2/issues/32810, it seems that we need to optimize the `lib/web/jquery/ui-modules/dialog.js` file.

But to speed up the processing of this issue, please provide us with the following information:

- Information on your environment
- Steps to reproduce
- Expected and actual results

Thanks

**Comment by engcom-Hotel** [comment]
Hello @onlinebizsoft,

Have you got a chance to check the above comment?

Thanks

**Comment by onlinebizsoft** [comment]
@engcom-Hotel 
> Information on your environment
> Steps to reproduce

I don't think we need any specific environment because I have mentioned clearly where the issue is wrong. I'm not sure in which case the dialog.js will be loaded (maybe it is loaded by default or you can declare a JS which use it)

> Expected and actual results

When dialog.js is used, it should not load TOO many other JS libs which is not necessary as in my screenshot.

You can see that the issue was fixed one time in past in ticket https://github.com/magento/magento2/issues/32810

**Comment by lbajsarowicz** [comment]
dialog.js now reports 'jQuery UI Dialog 1.14.2' after the AC-17370 upgrade from 1.14.1; the original complaint referenced a stale pre-upgrade commit (fa87f929). jQuery UI has been upgraded multiple times since this was filed. Please re-QA against the current jQuery UI version to confirm whether this still reproduces.
